src/frontend/Errors.ml

```diff
@@ -23,14 +22,21 @@ type t =
   | Semantic_error of Semantic_error.t
   | DebugDataError of (Middle.Location_span.t * string)
 
+let get_context_callback ?code filename () =
+  (* By the time we are printing an error, all these files are already resolved. *)
+  match !Include_files.include_provider with
+  | FileSystemPaths _ ->
+      (* So we can read directly from the filesystem *)
+      In_channel.read_lines filename
+  | InMemory m ->
+      (* Or it is either in the map, or, it is the original source file. *)
+      let included = Map.find m filename in
+      Option.first_some included code |> Option.value_exn |> String.split_lines
```

Comment:
Technically, this could be wrong if `filename-in-msg` also occurs in the includes list with different file content. (Of course, that's a very strange thing to do.)
I believe the correct logic would be that it's from the original source file when `loc.included_from` is `None` and from the map otherwise.